scripts/shieldscout.lua

```diff
@@ -170,25 +128,184 @@ end
 -----------------------------
 -- Walking
 
-local function Stopping()
-	Signal(SIG_Walk)
-	SetSignalMask(SIG_Walk)
-	
-	Move(pelvis, y_axis, 0.000000, 1.000000)
-	Turn(rightLeg.thigh, x_axis, 0, math.rad(200))
-	Turn(rightLeg.calf, x_axis, 0, math.rad(200))
-	Turn(rightLeg.foot, x_axis, 0, math.rad(200))
-	Turn(leftLeg.thigh, x_axis, 0, math.rad(200))
-	Turn(leftLeg.calf, x_axis, 0, math.rad(200))
-	Turn(leftLeg.foot, x_axis, 0, math.rad(200))
+local function GetSpeedMod()
+	local animFramesPerKeyframe = 4
+	return animFramesPerKeyframe / (Spring.GetUnitRulesParam(unitID, "totalMoveSpeedChange") or 1)
+end
+
+local animSpeed = GetSpeedMod()
+local walking = false
```

Comment:
Could use a comment to make sure the recent findings don't become tears in the rain
```suggestion
local walking = false -- required because StartMoving is spammed when stuck on spires, see #4251
```

Reply:
left a comment that it's to stop StartMoving getting spammed. The spire specific stuff seems related to jump, happens after the non somersault jump if unit was walking before (if jump is queued up outside of range so unit has to walk then jump)